assert.js

```diff
@@ -596,6 +596,51 @@ assert.ifError = function ifError(err) {
   }
 };
 
+// Currently in sync with Node.js lib/assert.js
+// https://github.com/nodejs/node/commit/2a871df3dfb8ea663ef5e1f8f62701ec51384ecb
+function internalMatch(string, regexp, message, fn) {
+  if (!isRegExp(regexp)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'regexp', 'RegExp', regexp
+    );
+  }
+  const match = fn.name === 'match';
+  if (typeof string !== 'string' ||
+      regexp.test(string) !== match) {
```

Comment:
this is vulnerable to `delete RegExp.prototype.test` (which node core is not); you'll want to call-bind that function at module level, like this:
```js
const test = require('call-bind/callBound')('RegExp.prototype.test');
```
and then you can do `test(regexp, string)` here.

Reply:
@ljharb Thanks, done!